### PR TITLE
Add support for VSCode

### PR DESCRIPTION
**Pull request by akhilveeraghanta**
### Description
Add support for VSCode

### Testing Done
- [x] Install script works and is repeatable
- [x] Code suggestion/refactoring/jump-to-def all works well
- [ ] Debugging :bug:, we have decided to not support debugging here, gdb should be used instead

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Comment by kelvinkoon** [comment]
:100: idea.

File: .vscode/extensions.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "recommendations": [
+        "ms-vscode.cpptools",
+        "bazelbuild.vscode-bazel",
```
Comment (jsn0716) [inline .vscode/extensions.json:4]: Did you manage to figure out why launching builds by clicking targets is broken? If not, I'm not sure what this extension is useful for - from their [README](https://github.com/bazelbuild/vscode-bazel), I guess Bazel Build Targets tree? Linting with buildifier is already done in `fix_formatting.sh`

**Review by jsn0716 — COMMENTED** [review]
LGTM, but I figured out how to get the click to build target working, you have to enable `Bazel: Enable Code Lens` in settings. Think this might be worth mentioning in `getting-started.md`?

**Review by jsn0716 — APPROVED** [review]
LGTM

**Review by jonathanlew — APPROVED** [review]

**Review by jordanzliu — APPROVED** [review]
works for me LGTM

**Pull request merged**